dags/test_docker_failure.py

```diff
@@ -0,0 +1,43 @@
+import os
+
+from airflow.models import DAG
+from airflow.operators.docker_operator import DockerOperator
+from pendulum import datetime, duration, now
+
+from utils.slack_operator import task_fail_slack_alert, slack_member_ids
+
+DEPLOYMENT_ENVIRONMENT = os.getenv("ENVIRONMENT", "development")
+
+DEFAULT_ARGS = {
+    "owner": "airflow",
+    "depends_on_past": False,
+    "start_date": datetime(2015, 1, 1, tz="America/Chicago"),
+    "email_on_failure": False,
+    "email_on_retry": False,
+    "retries": 0,
+    "execution_timeout": duration(minutes=5),
+    "on_failure_callback": task_fail_slack_alert,
+}
+
+with DAG(
+    dag_id=f"test_docker_failure",
+    description="Throws a python exception from within a docker container",
+    default_args=DEFAULT_ARGS,
+    schedule_interval=None,
+    tags=["repo:atd-airflow", "slack"],
+    catchup=False,
+) as dag:
+    dag.byline = f"{slack_member_ids['Frank']} test failure"
+    dag.icon = ":test_tube:"
+
+    t1 = DockerOperator(
+        task_id="docker_failure",
+        image="frankinaustin/pyfail:latest",
```

Comment:
This published image is defined later in this PR. It uses a modern python image and extends it to run a python program, included, that throws an exception.

Reply:
it'd be really nice if we could find a way to keep this test DAG around without depending on your image or keeping an image in our own repository. what do you think about this trickery?!

```python
    t1 = DockerOperator(
        task_id="docker_failure",
        image="atddocker/atd-airflow:production",
        command="python -c \"raise Exception('This is a test exception')\"",
        docker_conn_id="docker_default",
        auto_remove=True,
        tty=True,
        force_pull=True,
        mount_tmp_dir=False,
    )
```

Reply:
I just tested John's snippet ([slack link](https://austininnovation.slack.com/archives/C013G4RNJ01/p1740766358856229)), and it worked great! It would be a nice way to avoid depending on another image.

Reply:
I think this is neat trickery!!

Reply:
Much much better. 🙏  https://github.com/cityofaustin/atd-airflow/pull/270/commits/cbfdd611832c58ac9ffde2c7d7ccffda9607650a